monitors.go

```diff
@@ -152,3 +152,48 @@ func (m *Monitor) Instances(key *aurora.JobKey, instances int32, interval int, t
 	fmt.Println("Timed out")
 	return false, nil
 }
+
+// Monitor host status until all hosts match the status provided. May return an error along with a non nil map which contains
+// the hosts that did not transition to the desired modes(s).
+func (m *Monitor) HostMaintenance(hosts []string, modes []aurora.MaintenanceMode, sleepTime, steps int) (map[string]struct{}, error) {
```

Comment:
input to HostMaintenance is string list while return tuple has map[string]struct{}. Could have map[string]bool for return instead --  value "True" to indicate success and false otherwise..

Reply:
I gave this a decent amount of thought and came to the conclusion that  I'd be against it because we lose our O(1) way of checking wether we are done or now. Moving to a map[string]bool means we have to check the entire map every time we need to see if we're done, that means an O(N) search every step. This, however, might not be a problem until we're running at scale. 

On top of that, I feel this leaves us in about the same spot as we started because the user can take the non-transitioned hosts and check them against their original list with relative ease. What I feel would be an ideal enhancement here, as you pointed out earlier offline, would be finding which nodes passed in by the user don't exist. This brings forth three scenarios that I can see being troublesome:

1. Node is valid, however, it is temporarily missing from the list of nodes that Aurora has. This might lead to incorrectly marking down the node as non-existent. 
1. The node is valid originally, but goes away after we mark it as valid.
1. The node is not valid originally but joins the list of nodes Aurora is keeping track of while the monitoring is going on.

Let's see if we can come up with a way to at least tackle some of them.